I tensorflow/core/distributed_runtime/rpc/grpc_channel.cc:197] Initialize GrpcChannelCache for job ps -> {0 -> localhost:2222}
I tensorflow/core/distributed_runtime/rpc/grpc_channel.cc:197] Initialize GrpcChannelCache for job worker -> {0 -> 127.0.0.1:2222}
I tensorflow/core/distributed_runtime/rpc/grpc_server_lib.cc:206] Started server with target: grpc://localhost:2222
I tensorflow/core/distributed_runtime/rpc/grpc_channel.cc:197] Initialize GrpcChannelCache for job worker -> {0 -> localhost:2223}
I tensorflow/core/distributed_runtime/rpc/grpc_server_lib.cc:206] Started server with target: grpc://localhost:2223
Traceback (most recent call last):
  File "tfserver.py", line 89, in <module>
    tf.app.run()
  File "/root/tensorflow/local/lib/python2.7/site-packages/tensorflow/python/platform/app.py", line 30, in run
    sys.exit(main(sys.argv[:1] + flags_passthrough))
  File "tfserver.py", line 46, in main
    import train_2 as mtrain
  File "/home/pan/bitcoin tf/train_2.py", line 31, in <module>
    sess.run(tf.initialize_all_variables())
  File "/root/tensorflow/local/lib/python2.7/site-packages/tensorflow/python/client/session.py", line 717, in run
    run_metadata_ptr)
  File "/root/tensorflow/local/lib/python2.7/site-packages/tensorflow/python/client/session.py", line 915, in _run
    feed_dict_string, options, run_metadata)
  File "/root/tensorflow/local/lib/python2.7/site-packages/tensorflow/python/client/session.py", line 965, in _do_run
    target_list, options, run_metadata)
  File "/root/tensorflow/local/lib/python2.7/site-packages/tensorflow/python/client/session.py", line 985, in _do_call
    raise type(e)(node_def, op, message)
tensorflow.python.framework.errors.InvalidArgumentError: Cannot assign a device to node 'W_add': Could not satisfy explicit device specification '/job:ps/task:0' because no devices matching that specification are registered in this process; available devices: /job:localhost/replica:0/task:0/cpu:0
Colocation Debug Info:
Colocation group had the following types and devices: 
Assign: CPU 
Variable: CPU 
	 [[Node: W_add = Variable[container="", dtype=DT_FLOAT, shape=[1], shared_name="", _device="/job:ps/task:0"]()]]

Caused by op u'W_add', defined at:
  File "tfserver.py", line 89, in <module>
    tf.app.run()
  File "/root/tensorflow/local/lib/python2.7/site-packages/tensorflow/python/platform/app.py", line 30, in run
    sys.exit(main(sys.argv[:1] + flags_passthrough))
  File "tfserver.py", line 46, in main
    import train_2 as mtrain
  File "/home/pan/bitcoin tf/train_2.py", line 5, in <module>
    import model
  File "/home/pan/bitcoin tf/model.py", line 162, in <module>
    W_add = bias_variable([1],name='W_add')
  File "/home/pan/bitcoin tf/model.py", line 74, in bias_variable
    return tf.Variable(initial,name=name)
  File "/root/tensorflow/local/lib/python2.7/site-packages/tensorflow/python/ops/variables.py", line 215, in __init__
    dtype=dtype)
  File "/root/tensorflow/local/lib/python2.7/site-packages/tensorflow/python/ops/variables.py", line 300, in _init_from_args
    name=name)
  File "/root/tensorflow/local/lib/python2.7/site-packages/tensorflow/python/ops/state_ops.py", line 146, in variable_op
    container=container, shared_name=shared_name)
  File "/root/tensorflow/local/lib/python2.7/site-packages/tensorflow/python/ops/gen_state_ops.py", line 490, in _variable
    name=name)
  File "/root/tensorflow/local/lib/python2.7/site-packages/tensorflow/python/framework/op_def_library.py", line 749, in apply_op
    op_def=op_def)
  File "/root/tensorflow/local/lib/python2.7/site-packages/tensorflow/python/framework/ops.py", line 2380, in create_op
    original_op=self._default_original_op, op_def=op_def)
  File "/root/tensorflow/local/lib/python2.7/site-packages/tensorflow/python/framework/ops.py", line 1298, in __init__
    self._traceback = _extract_stack()

InvalidArgumentError (see above for traceback): Cannot assign a device to node 'W_add': Could not satisfy explicit device specification '/job:ps/task:0' because no devices matching that specification are registered in this process; available devices: /job:localhost/replica:0/task:0/cpu:0
Colocation Debug Info:
Colocation group had the following types and devices: 
Assign: CPU 
Variable: CPU 
	 [[Node: W_add = Variable[container="", dtype=DT_FLOAT, shape=[1], shared_name="", _device="/job:ps/task:0"]()]]

F tensorflow/core/common_runtime/device.cc:29] Check failed: DeviceNameUtils::ParseFullName(name(), &parsed_name_) Invalid device name: /job:/replica:0/task:0/cpu:0
F tensorflow/core/common_runtime/device.cc:29] Check failed: DeviceNameUtils::ParseFullName(name(), &parsed_name_) Invalid device name: /job:/replica:0/task:0/cpu:0
F tensorflow/core/common_runtime/device.cc:29] Check failed: DeviceNameUtils::ParseFullName(name(), &parsed_name_) Invalid device name: /job:/replica:0/task:0/cpu:0
I tensorflow/core/distributed_runtime/rpc/grpc_channel.cc:197] Initialize GrpcChannelCache for job ps -> {0 -> localhost:2222}
I tensorflow/core/distributed_runtime/rpc/grpc_channel.cc:197] Initialize GrpcChannelCache for job worker -> {0 -> 127.0.0.1:2222}
I tensorflow/core/distributed_runtime/rpc/grpc_server_lib.cc:206] Started server with target: grpc://localhost:2222
I tensorflow/core/distributed_runtime/rpc/grpc_channel.cc:197] Initialize GrpcChannelCache for job ps -> {0 -> localhost:2222}
I tensorflow/core/distributed_runtime/rpc/grpc_channel.cc:197] Initialize GrpcChannelCache for job worker -> {0 -> 127.0.0.1:2222}
I tensorflow/core/distributed_runtime/rpc/grpc_server_lib.cc:206] Started server with target: grpc://localhost:2222
